packages/hive/sodasql/dialects/hive_dialect.py

```diff
@@ -47,6 +47,15 @@ def default_connection_properties(self, params: dict):
             'database': params.get('database', 'your_database')
         }
 
+    def safe_connection_data(self):
+        return [
+            self.type,
+            self.host,
```

Comment:
we will push this into `datasource_id`

Reply:
not sure what you mean, but this list is what goes to our hashing function that _generates_ the `datasource_id`. The hashing is (arguably) irreversible, so no one will be able to determine host/user/schema or any of these `safe_connection_data` values from the hash. We add all this info as input to the hashing so that we have enough unique values coming into it to ensure uniqueness of the hash. Otherwise we would generate conflicting `datasource_ids` (hashing is deterministic, i.e. we would end up with same hashes for what would actually be different warehouses)

Reply:
I think we can be ok with the following:
- host
- port

we will have the type already as `datasource_type` anyway and it doesn't make sense to me to include `database` (since our definition of core system is that they use some host, if things happen in separate dbs and what not we don't really care to differentiate.

In the unlikely event that someone was table to reverse it they would have some credentials + some proprietary info. It's not uncommon for database to carry some info about what is in there and users may not want us to be able to know.

From an analytical point of view host is enough host and port is also fine, and I would argue we don't even need port.

Reply:
I can buy that @m1n0! I totally understand where you were coming from but wanted to push to the point where we think that we're tracking the correct components. Your point about potential generic host names is very good. I actually made a comment where I was voicing that same concern in the form of a question somewhere.

And I agree that if we think the host is likely to be often the same then type could add some entropy.

The username, however, besides being potentially a security risk, may have added too much entropy in the case where several users connect to the same host and type but with a different username. I think this scenario is likely and it would have actually prevented us of getting an idea of a "team, using a core system" do you see what I mean?

So the decision to remove it solves those two issues and is a good decision in my opinion.